src/extensions/surveys.tsx

```diff
@@ -99,42 +89,43 @@ export const callSurveys = (posthog: PostHog, forceReload: boolean = false) => {
 
                 if (!localStorage.getItem(`seenSurvey_${survey.id}`)) {
                     const shadow = createShadow(style(survey?.appearance), survey.id)
-                    Preact.render(<Surveys key={'popover-survey'} posthog={posthog} survey={survey} />, shadow)
+                    Preact.render(<SurveyPopup key={'popover-survey'} posthog={posthog} survey={survey} />, shadow)
                 }
             }
         })
     }, forceReload)
 }
 
-export const renderSurveysPreview = (
-    survey: Survey,
-    root: HTMLElement,
-    displayState: 'survey' | 'confirmation',
-    previewQuestionIndex: number
-) => {
+export const renderSurveysPreview = (survey: Survey, parentElement: HTMLElement, previewPageIndex: number) => {
     const surveyStyleSheet = style(survey.appearance)
     const styleElement = Object.assign(document.createElement('style'), { innerText: surveyStyleSheet })
-    root.appendChild(styleElement)
+
+    // Remove previously attached <style>
+    Array.from(parentElement.children).forEach((child) => {
```

Comment:
Why do we need to remove and then add a new generated style tag here?

It also doesn't seem like the `survey.appearance` can change in the middle. Maybe all you need is `useMemo` here to keep the original generated element

Reply:
It _can_ change. Any changes made in the SurveyEdit form are propagated via `survey.appearance` which is then converted into a stylesheet and attached.

The survey popup is re-rendered on the same parent element, which still has the previous stylesheet, which must be removed.

Reply:
Do you mean when I make a change in the UI? In that case, we can still `useMemo` which listens on `survey.appearance` and only updates the stylesheet when the appearance changes.

Reply:
Maybe I'm missing something here but I don't follow 🤔 this code is part of a rendering method `renderSurveysPreview`, not a React component, so `useMemo` can't be used here.

Anytime a popup is adjusted in the UI, we grab the root (parent) element and render a new popup to it. However, `renderSurveysPreview` also attaches a stylesheet to the parent, and so each invocation of this method will add a new stylesheet. Thus we want to remove any previously added stylesheets.

Reply:
d'oh, I haven't been reading properly 😅 - makes sense